modules/externalipam/netbox/netbox_client.rb

```diff
@@ -0,0 +1,201 @@
+      response = @api_resource.get("ipam/prefixes/?#{params}")
+      json_body = JSON.parse(response.body)
+      return nil if json_body['count'].zero?
+      subnet = subnet_from_result(json_body['results'][0])
+      return subnet if json_body['results']
+    end
+
+    def get_ipam_subnet_by_cidr(cidr)
+      params = URI.encode_www_form({ status: 'active', prefix: cidr })
+      response = @api_resource.get("ipam/prefixes/?#{params}")
+      json_body = JSON.parse(response.body)
+      return nil if json_body['count'].zero?
+      subnet = subnet_from_result(json_body['results'][0])
+      return subnet if json_body['results']
+    end
+
+    def ipam_groups
+      response = @api_resource.get('ipam/vrfs/')
+      json_body = JSON.parse(response.body)
+      groups = []
+
+      return groups if json_body['count'].zero?
+
+      json_body['results'].each do |group|
+        groups.push(
+          {
+            name: group['name'],
+            description: group['description'],
+          }
+        )
+      end
+
+      groups
+    end
+
+    def get_ipam_group(group_name)
+      raise ERRORS[:groups_not_supported] unless groups_supported?
+      response = @api_resource.get("ipam/vrfs/?name=#{group_name}")
+      json_body = JSON.parse(response.body)
+      return nil if json_body['count'].zero?
+
+      group = {
+        id: json_body['results'][0]['id'],
+        name: json_body['results'][0]['name'],
+        description: json_body['results'][0]['description'],
+      }
+
+      return group if json_body['results']
+    end
+
+    def get_group_id(group_name)
+      return nil if group_name.nil? || group_name.empty?
+      group = get_ipam_group(group_name)
+      raise ERRORS[:no_group] if group.nil?
+      group[:id]
+    end
+
+    def get_ipam_subnets(group_name)
+      if group_name.nil?
+        params = URI.encode_www_form({ status: 'active' })
+      else
+        group_id = get_group_id(group_name)
+        params = URI.encode_www_form({ status: 'active', vrf_id: group_id })
+      end
+
+      response = @api_resource.get("ipam/prefixes/?#{params}")
+      json_body = JSON.parse(response.body)
+      return nil if json_body['count'].zero?
+      subnets = []
+
+      json_body['results'].each do |subnet|
+        subnets.push(
+          {
+            subnet: subnet['prefix'].split('/').first,
+            mask: subnet['prefix'].split('/').last,
+            description: subnet['description'],
+            id: subnet['id'],
+          }
+        )
+      end
+
+      return subnets if json_body['results']
+    end
+
+    def ip_exists?(ip, subnet_id, group_name)
+      group_id = get_group_id(group_name)
+      url = "ipam/ip-addresses/?#{URI.encode_www_form({ address: ip })}"
+      url += "&#{URI.encode_www_form({ prefix_id: subnet_id })}" unless subnet_id.nil?
+      url += "&#{URI.encode_www_form({ vrf_id: group_id })}" unless group_id.nil?
```

Comment:
What I would write is:
```suggestion
      params = {address: ip}
      params[:prefix_id] = subnet_id unless subnet_id.nil?
      params[:vrf_id] = group_id unless group_id.nil?
      url = "ipam/ip-addresses/?#{URI.encode_www_form(params)}"
```